File: python/leetcode/1_easy/_0136_single_number.py
class Solution:
    def singleNumber(self, nums: list[int]) -> int:
        # result = 0

        # for num in nums:
        #     result ^= num
        # return result
        

        n = len(nums)
        for i in range(n):
            count = 0
            for j in range(n):
                if nums[i] == nums[j]:
                    count += 1
            if count == 1:
                return nums[i]
            
        return -1

def main():
    s = Solution()
    print(s.singleNumber([4,1,2,1,2]))
    print(s.singleNumber([2, 2, 1]))

if __name__ == "__main__":
    main()

File: rust/src/_1_easy/_0136_single_number.rs
#[allow(dead_code)]
pub mod solution {
    pub struct Solution;

    impl Solution {
        pub fn single_number(nums: Vec<i32>) -> i32 {
            let mut result = 0;

            for num in nums {
                result ^= num;
            }

            return result
        }
    }
}

#[allow(dead_code)]
pub fn run() {
    use solution::Solution;
    let result = Solution::single_number(vec![2, 2, 1]);
    println!("{:?}", result);
}